src/sentry/scim/endpoints/utils.py

```diff
@@ -160,6 +161,23 @@ def get_query_parameters(self, request: Request):
 
         return serializer.validated_data
 
+    def fix_log_name(self, request: Request):
+        # fix username if needed
+        scim_prefix = "scim-internal-integration-"
+        scim_regex = re.compile(
+            scim_prefix
+            + r"[0-9a-fA-F]{6}\-[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{7}"
+        )
+        scim_match = re.match(scim_regex, request.user.username)
+
+        if scim_match:
+            uuid = request.user.username.split(scim_prefix)[1]
+            uuid_prefix = uuid.split("-")[0]
```

Comment:
This is fine with the `split`s, but it could also be rewritten to use [capturing groups](https://meet.google.com/ias-zfir-uuo) if you want to be a little more elegant.